apiserver/params/controller.go

```diff
@@ -132,10 +136,138 @@ type ControllerVersionResults struct {
 	GitCommit string `json:"git-commit"`
 }
 
-// DashboardInfo holds the results from an api call
-// to get address info for the juju dashboard.
-type DashboardInfo struct {
-	Addresses []string `json:"addresses"`
-	UseTunnel bool     `json:"use-tunnel"`
-	Error     *Error   `json:"error,omitempty"`
+const (
+	// DashboardConnectionTypeProxy is the type key used for proxy connections
+	DashboardConnectionTypeProxy = "proxy"
+
+	// DashboardConnectionTypeSSHTunnel is the type key used for ssh connections
+	DashboardConnectionTypeSSHTunnel = "ssh-tunnel"
+)
+
+// DashboardConnectionProxy represents a proxy connection to the Juju Dashboard
+type DashboardConnectionProxy struct {
+	Proxier       proxy.Proxier   `json:"proxier"`
```

Comment:
We shouldn't be passing complex types like this across the wire. What needs to go across is any necessary config needed to construct the proxier on the calling side. Otherwise we run into all sorts of compatibility issues.
The params structs are for structs with attributes, ie DTOs